packages/app-store/googlecalendar/api/callback.ts

```diff
@@ -19,6 +20,13 @@ async function getHandler(req: NextApiRequest, res: NextApiResponse) {
   const state = decodeOAuthState(req);
 
   if (typeof code !== "string") {
+    if (state?.onErrorReturnTo) {
+      res.redirect(
+        getSafeRedirectUrl(state.onErrorReturnTo) ??
+          getSafeRedirectUrl(state?.returnTo) ??
+          `${CAL_URL}/apps/installed`
+      );
+    }
     throw new HttpError({ statusCode: 400, message: "`code` must be a string" });
```

Comment:
Even when we redirect the user back onError - we are still throwing this HttpError - I think this needs to be in the above `else` statement.

Reply:
I was getting weird type errors if I tried putting it in an `else` block. How do you feel about adding return statements? cfce15582317a5e32f4dcb891d380122b835ccd6

Reply:
Fine by me :)